Fix sass import deprecation

robsoncombr opened the pull request:
## Scope

I was able to build it without any error, I have tested and I was not able to find any problem, the deprecation warning is also gone. This was really challenging, I'm not sure I did everything correctly and I don't think I can certify the standard from here on my own, etc. I would appreciate if you could help me. Thanks.

### Find the success build below:

```
* Executing task in folder ubuntu: docker exec -it 4fca7d354821b5a1dda9fb2c7a30c084d75d62c8d3ec595ccb204bd68f682d5d bash

I have no name!@4fca7d354821:/directus$ pnpm --filter app build

@directus/app@13.3.6 build /directus/app
vite build

vite v5.4.10 building for production...
✓ 4422 modules transformed.
[plugin:vite:reporter] [plugin vite:reporter]
(!) /directus/node_modules/.pnpm/date-fns@4.1.0/node_modules/date-fns/locale/en-US.js is dynamically imported by /directus/app/src/utils/get-date-fns-locale.ts but also statically imported by /directus/node_modules/.pnpm/date-fns@4.1.0/node_modules/date-fns/_lib/defaultLocale.js, dynamic import will not move module into another chunk.

[plugin:vite:reporter] [plugin vite:reporter]
(!) /directus/app/src/lang/translations/en-US.yaml is dynamically imported by /directus/app/src/lang/set-language.ts but also statically imported by /directus/app/src/lang/index.ts, dynamic import will not move module into another chunk.

[plugin:vite:reporter] [plugin vite:reporter]
(!) /directus/node_modules/.pnpm/codemirror@5.65.18/node_modules/codemirror/mode/markdown/markdown.js is dynamically imported by /directus/app/src/interfaces/input-code/import-codemirror-mode.ts but also statically imported by /directus/node_modules/.pnpm/codemirror@5.65.18/node_modules/codemirror/mode/markdown/markdown.js?commonjs-proxy, /directus/app/src/interfaces/input-rich-text-md/input-rich-text-md.vue?vue&type=script&setup=true&lang.ts, dynamic import will not move module into another chunk.

[plugin:vite:reporter] [plugin vite:reporter]
(!) /directus/node_modules/.pnpm/codemirror@5.65.18/node_modules/codemirror/mode/javascript/javascript.js is dynamically imported by /directus/app/src/interfaces/input-code/import-codemirror-mode.ts, /directus/app/src/interfaces/input-code/input-code.vue?vue&type=script&setup=true&lang.ts but also statically imported by /directus/node_modules/.pnpm/codemirror@5.65.18/node_modules/codemirror/mode/javascript/javascript.js?commonjs-proxy, /directus/app/src/modules/settings/routes/system-logs/logs.vue?vue&type=script&setup=true&lang.ts, dynamic import will not move module into another chunk.

dist/index.html 1.83 kB │ gzip: 0.74 kB
dist/assets/FiraMono-Medium-CE-6Frqh.woff2 16.15 kB
(.removed.)
dist/assets/index.yA6csgyz.entry.js 6,013.85 kB │ gzip: 1,755.80 kB

(!) Some chunks are larger than 500 kB after minification. Consider:

Using dynamic import() to code-split the application
Use build.rollupOptions.output.manualChunks to improve chunking: https://rollupjs.org/configuration-options/#output-manualchunks
Adjust chunk size limit for this warning via build.chunkSizeWarningLimit.
✓ built in 1m 10s
```

## Potential Risks / Drawbacks

- With this change in place new scss as file or inside vue compoenents should follow new `@use` pattern
- **I consider that it is indispensable that a Senior Front-End review this**, there is a potential chance I did something wrong or not good enough, all I have for now is the building with no error and some fast running tests in browser.

## Review Notes / Questions

- With this change in place new scss as file or inside vue compoenents should follow new `@use` pattern
- I think that because of the pattern change we should have a Changeset bumped on the app, I would also like to have assistance on this matter, thanks.

---

Fixes #24198

robsoncombr committed:
patch: temp fix for issue 9521, to be reverted with permanent solution

robsoncombr committed:
Merge branch 'directus:main' into main

robsoncombr committed:
Merge branch 'directus:main' into main

robsoncombr committed:
Merge branch 'directus:main' into main

robsoncombr committed:
Merge branch 'directus:main' into main

robsoncombr committed:
fix 24198 sass import deprecation

robsoncombr committed:
fix: manually fixing issues after run sass-migrator

robsoncombr committed:
fix: some more fix, not build success, need to fix the .vue components, @import '@/styles...'

robsoncombr committed:
Merge branch 'directus:main' into fix-24198-sass-import-deprecation

robsoncombr committed:
fix: moving @import to @use pattern inside .vue compoenents

robsoncombr committed:
fix: finished to fix the final errors, will really need a bless on this

robsoncombr committed:
fix: rollback oauth2 api driver, probably changed by mistake while opened in vscode

robsoncombr committed:
Merge branch 'main' into fix-24198-sass-import-deprecation

robsoncombr committed:
fix: prettier formatter

robsoncombr committed:
Merge branch 'fix-24198-sass-import-deprecation' of https://github.com/robsoncombr/directus into fix-24198-sass-import-deprecation

robsoncombr committed:
Merge branch 'main' into fix-24198-sass-import-deprecation

rijkvanzanten commented:
Holy smokes friend, you've been busy! I saw the errors when @paescuj did his Update Vite PR #23845 and even tried migrating some parts, but gave up as there were a lot of edge cases and it isn't deprecated _yet_ 🫠 Thanks for this!

robsoncombr committed:
Merge branch 'main' into fix-24198-sass-import-deprecation

robsoncombr commented:
@rijkvanzanten 
_> Holy smokes friend, you've been busy! I saw the errors when @paescuj did his Update Vite PR #23845 and even tried migrating some parts, but gave up as there were a lot of edge cases and it isn't deprecated _yet_ 🫠 Thanks for this!_

I’m very happy to help. When I found Directus, very recently, I simply found the project I'm designing for the last 20 years, but never managing to finish. Three years ago, I wrote some cool code in my newest implementation try and my vision has always been very close to what Directus offers today, have some other ideas that I can share some day.

Also, while I still have your eyes, just to mention, I'm still new and arriving, but looking around, today I noticed discussions about noSQL / MongoDB support implementations #3156. So I wanted to mention that during this last 'self-Directus' implementation try I mentioned, I was using MongoDB and I created 2 nice solutions that I can share with you guys when it is time to work on this MongoDB matter.

- One is a simple workaround for the lack of native full-search feature, that can certainly be improved but might be useful as a beginning or at least for some particular collections, since it does not depend on external tools.
- The other, a microservice I wrote in Node/Express, this piece was very nice, that based on the oplog allow us to implement automations, under the roadmap I had: data streaming, realtime, audit records, backups, revisions, triggers. I implemented a safe-resume to avoid losing any transaction if service is restored from any down time event.

Who knows, maybe some time I create a repository to open-source it, for sure might be useful for someone.

Well, gotta go for today, I hope I’ve been helpful, if you have any questions, just let me know!

# Fly high, Directus! 🚀

connorwinston committed:
Merge branch 'main' into fix-24198-sass-import-deprecation

connorwinston reviewed (dismissed):
Wow!! Thanks for this PR @robsoncombr. From my testing this all works, just a few things I left comments on.

Since `@use` now imports stuff under a namespace, lets make a index file in `@/style/mixins/` that uses `@forward` to forward all the mixins in that folder under a `mixin` namespace. So that way we can use it like the below.

```scss
@use '@/styles/mixins';

.example-class {
	@include mixins.form-grid;
}
```

Example of `@/styles/mixins/index.scss`
```scss
@forward "form-grid";
@forward "markdown";
@forward "no-wrap";
@forward "type-styles";
```
Let me know if you have any questions!

AlexGaillard commented:
Thanks for this @robsoncombr 🚀  We're hoping to get this into the next release. So we're going to hand it over internally for someone to finish off if you haven't started with @connorwinston's feedback yet?

robsoncombr committed:
Merge branch 'main' into fix-24198-sass-import-deprecation

robsoncombr commented:
> Thanks for this @robsoncombr 🚀 We're hoping to get this into the next release. So we're going to hand it over internally for someone to finish off if you haven't started with @connorwinston's feedback yet? 😄

Hey, I'm sorry, I got a little busy and lost this conversation.. I will review now.

robsoncombr commented on app/src/styles/themes/mixins/generate-colors.scss:
```diff
@@ -1,22 +1,21 @@
 @mixin generate-colors($color-map, $mix-color-low, $mix-color-high) {
-	// Loop through palette in $color-map list
 	@each $identifier, $color in $color-map {
-		// Loop through pre-defined steps in $step-mapping list
-		@each $suffix, $step in $step-mapping {
-			// Default suffix to nothing
+		@each $suffix, $step in colors.$step-mapping {
 			$resolved-suffix: '';
-			// Default mixing color
 			$mixing-color: $mix-color-low;
-			// Set suffix if not equal to none
+
 			@if $suffix != 'none' {
 				$resolved-suffix: '-#{$suffix}';
 			}
-			// If our step is negative, change to dark mixing color
 			@if $step < 0 {
 				$mixing-color: $mix-color-high;
 			}
-			// This is the CSS variable that will be output
```
Nope, no reason from my part, it was remove by my local pre-commit processor, I will disable and revert, thanks for pointing it out!

robsoncombr commented on app/src/styles/main.scss:
```diff
@@ -1,25 +1,25 @@
+@use './base';
+@use './colors' as colors;
+@use './fonts';
+@use './themes/dark' as dark;
+@use './themes/light' as light;
```
Gotcha, thanks for the guidance! 😎

robsoncombr committed:
Update app/src/styles/main.scss

Better namespacing naming

Co-authored-by: Connor Winston <connor@winston.guru>

robsoncombr committed:
fix: added comments back in

robsoncombr committed:
Update app/src/styles/main.scss

applying more explicit namespace guideline

Co-authored-by: Connor Winston <connor@winston.guru>

robsoncombr committed:
Update app/src/styles/main.scss

applying more explicit namespace guideline

Co-authored-by: Connor Winston <connor@winston.guru>

robsoncombr commented on app/src/styles/themes/_dark.scss:
```diff
@@ -27,7 +30,7 @@
 
 	// Generate color variations
-	@include generate-colors($color-mapping, $dark-theme-shade, $dark-theme-tint);
+	@include generate-colors.generate-colors($color-mapping, colors.$dark-theme-tint, colors.$dark-theme-shade);
```
I had a look into my file changes history, I don´t remember to manually change this, good caught, my only theory here is that this was changed by sass-migrator tool, that after experience with this PR, I can say it does not work well, no way to migrate without a lot of manual interventions. Anyway, I will revert it, I'm also a fan of not applying changes that aren't directly related to the task. 😎

robsoncombr committed:
Update app/src/styles/themes/_dark.scss

applying the change request to avoid unecessary changes made by automated tool sass-migrate.

Co-authored-by: Connor Winston <connor@winston.guru>

robsoncombr commented:
Ok mates, sorry for the delay, I have reviwed all requests and we are good now.
Let me know any further observation you may have!
Thanks for backing me up. 🤘

formfcw committed:
refactor SCSS mixins usage for consistency and clarity

formfcw committed:
bring back removed comments

formfcw committed:
remove unused SCSS imports

formfcw commented:
@robsoncombr Thank you :rocket: 

I added commits to resolve Connor's request from above https://github.com/directus/directus/pull/24201#pullrequestreview-2510279649 and did some cleanup.

@AlexGaillard Maybe someone can approve only my commits, since the rest LGTM  🎉

paescuj committed:
Consistent import URLs

paescuj committed:
Use index for themes

paescuj committed:
Consistent file naming (partials)

paescuj committed:
Use (consistent) direct imports for external CSS

paescuj committed:
Migrate import in histoire wrapper

paescuj reviewed (approved):
Awesome work, thank you all ❤️
